File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {

        while(head != NULL && head->val == val){
            cout<<head->val<<" ";
            head = head->next;
        }
        if(head == NULL) return NULL;

        ListNode* prev = NULL;
        ListNode* mover = head;

        while(mover != NULL){
            if(mover->val == val){
                prev->next = prev->next->next;
                mover = mover->next;
                continue;
            }
            cout<<mover->val<<" ";
            prev = mover;
            mover = mover->next;
        }
        return head;

    }
};

File: 0203-remove-linked-list-elements/0203-remove-linked-list-elements.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:
        if head == None: return head

        while(head != None and head.val == val):
            head = head.next
        if head == None: return head

        temp = head; prev = None
        while temp != None:
            if(temp.val == val): 
                prev.next = prev.next.next
                temp = temp.next
                continue
            prev = temp
            temp = temp.next
        
        return head
